app/src/androidTest/java/org/simple/clinic/drugs/PrescriptionRepositoryAndroidTest.kt

```diff
@@ -111,6 +111,29 @@ class PrescriptionRepositoryAndroidTest {
     assertThat(savedPrescriptions2).hasSize(1)
   }
 
+  @Test
+  fun soft_delete_prescription_should_update_timestamp_and_sync_status() {
+    val protocolUUID = UUID.randomUUID()
+    val patientUUID = UUID.randomUUID()
+
+    val drug = ProtocolDrug(UUID.randomUUID(), name = "Amlodipine", rxNormCode = null, dosages = listOf("5mg", "10mg"), protocolUUID = protocolUUID)
+
+    repository.savePrescription(patientUUID, drug, "5mg").blockingAwait()
+
+    val savedPrescriptions = repository.newestPrescriptionsForPatient(patientUUID).blockingFirst()
+    assertThat(savedPrescriptions).hasSize(1)
+
+    val prescription = savedPrescriptions[0]
+
+    repository.softDeletePrescription(prescription.uuid).blockingAwait()
+
+    val softDeletedPrescription = database.prescriptionDao().getOne(prescription.uuid)!!
+
+    assertThat(softDeletedPrescription.updatedAt).isGreaterThan(prescription.updatedAt)
```

Comment:
We were using `isGreaterThan` in another android test and we found that the two timestamps were occasionally the same because the code executes so fast. Rare case, but something to keep in mind.

Reply:
Does it? `Instant` takes the `nanos` into account as well, so those must have been some really fast tests.

Here, we are testing whether the older prescriptions `updatedAt` and the soft deleted prescriptions `updatedAt` are different(soft deleted < older). Do you feel it would be likely for this case to be triggered? What do you suggest we use instead? Because we do have to test whether the timestamp actually changes or not.

Reply:
We'll keep an eye out and figure an alternative if we see it happen then.